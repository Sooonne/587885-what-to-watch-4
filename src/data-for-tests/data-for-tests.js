export const PROMO_FILM = {
  title: `Star Wars`,
  genre: `Drama`,
  release: 2017,
  img: `img/the-grand-budapest-hotel.jpg`,
  bg: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 8.9,
  ratingCount: 240,
  id: 9,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

export const MOVIES = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 1,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 10,
    isFaborite: false,
  },
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 2,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 20,
    isFaborite: false,
  },
  {
    title: `Macbeth`,
    genre: `Comedy`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 3,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 0,
    isFaborite: false,
  },
  {
    title: `Aviator`,
    genre: `Comedy`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 4,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 40,
    isFaborite: false,
  },
  {
    title: `We need to talk about Kevin`,
    genre: `Thriller`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 5,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 50,
    isFaborite: false,
  },
  {
    title: `What We Do in the Shadows`,
    genre: `Thriller`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 6,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 60,
    isFaborite: false,
  },
  {
    title: `Pulp Fiction`,
    genre: `Crime`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 7,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 70,
    isFaborite: false,
  },
  {
    title: `Snatch`,
    genre: `Crime`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 8,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 80,
    isFaborite: false,
  }
];


export const MOVIE = {
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  genre: `Drama`,
  release: 2017,
  img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  bg: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 8.9,
  ratingCount: 240,
  id: 1,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

export const VIDEO = {
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};


export const REVIEW = {
  id: 1,
  user: {
    id: 2,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`
};

export const REVIEWS = [
  {
    id: 1,
    user: {
      id: 2,
      name: `Kate Muir`,
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 8,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 4,
    user: {
      id: 1,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 5,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
];
