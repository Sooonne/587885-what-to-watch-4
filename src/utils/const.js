export const PAGE_NAMES = {
  INFO: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const PAGES = {
  MAIN: `main`,
  MOVIE: `movie`,
};

export const ALL_GENRES = `All genres`;
export const MAX_GENRES_AMOUNT = 10;

export const extend = (state, newStateValue) => {
  return Object.assign({}, state, newStateValue);
};

export const filterMoviesByGenre = (allMovies, activeGenre) => {
  if (activeGenre === ALL_GENRES) {
    return allMovies;
  }
  return allMovies.filter((it) => it.genre === activeGenre);
};

export const getReviewsForMovie = (allReviews, movie) => {
  return allReviews.filter((it) => it.id === movie.genre);
};

export const getAllGenres = (allMovies) => {
  const genres = new Set(allMovies.map((it) => it.genre));
  return [ALL_GENRES].concat(...genres);
};

export const getMaxGenres = (genres) => {
  return genres.slice(0, MAX_GENRES_AMOUNT);
};

