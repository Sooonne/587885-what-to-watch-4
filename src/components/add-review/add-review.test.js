import React from "react";
import renderer from "react-test-renderer";
import AddReview from "./add-review.jsx";
import configureStore from "redux-mock-store";
import {PROMO_FILM, MOVIES} from "../../data-for-tests/data-for-tests";
import {Provider} from "react-redux";
import NameSpace from '../../reducer/name-space';
import {AuthorizationStatus} from "../../utils/const.js";
import {BrowserRouter as Router} from 'react-router-dom';

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.DATA]: {
    movieCard: PROMO_FILM,
    movies: MOVIES,
    submitStatus: `Default`,
  },
  [NameSpace.APP]: {
    filteredMovies: MOVIES,
    genres: [`All genres`],
    activeGenre: `All genres`,
    showedMoviesAmount: 8,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
    userInfo: {
      id: 1,
      email: `1@1.ru`,
      name: `Yo`,
      avatarUrl: `/pic1.jpg`
    },
  }
});

it(`Render App`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router>
            <AddReview
              movieCard = {PROMO_FILM}
              submitStatus = {`Default`}
              onSubmitClick = {() => {}}
              onRatingChange = {() => {}}
              onReviewChange = {() => {}}
              isSubmitDisabled = {true}
            />
          </Router>
        </Provider>
        , {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
