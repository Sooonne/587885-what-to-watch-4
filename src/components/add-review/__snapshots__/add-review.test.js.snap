// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/9"
              onClick={[Function]}
            >
              Star Wars
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <a
        className="user-block"
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="Yo"
            height="63"
            src="/pic1.jpg"
            width="63"
          />
        </div>
      </a>
      <div
        className="user-block"
      >
        <button
          className="user-block__link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Star Wars"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          onChange={[Function]}
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          maxLength={400}
          minLength={50}
          name="comment"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
